### PR TITLE
Fix MySQL error when course has no lessons (new PR)

### LINKED ISSUE #2012: SQL Error on reporting course overview when course has no lessons
### Reproduction Steps

1. Create a course without any lessons.
2. Navigate to Reporting -> Courses -> select the course without lessons

Fixed by #1875.

### Expected Behavior

No SQL error is displayed or logged to the PHP error log.

### Actual Behavior

An error message is displayed and logged to the PHP error log.

### Error Messages / Logs

+ Include any relevant error messages or log files
```
WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')
			  AND updated_date BETWEEN '2022-02-22 00:00:00' AND '2022-02-22 23:59:59'' at line 5 for query 
			SELECT COUNT( * )
			FROM wp_lifterlms_user_postmeta
			WHERE meta_value = 'yes'
			  AND meta_key = '_is_complete'
			  AND post_id IN (  )
			  AND updated_date BETWEEN '2022-02-22 00:00:00' AND '2022-02-22 23:59:59'
			 made by do_action('lifterlms_page_llms-reporting'), WP_Hook->do_action, WP_Hook->apply_filters, LLMS_Admin_Menus->reporting_page_init, LLMS_Admin_Reporting->output, llms_get_template, include('/plugins/lifterlms/templates/admin/reporting/reporting.php'), do_action('llms_reporting_content_courses'), WP_Hook->do_action, WP_Hook->apply_filters, LLMS_Admin_Reporting_Tab_Courses->output, llms_get_template, include('/plugins/lifterlms/templates/admin/reporting/tabs/courses/course.php'), llms_get_template, include('/plugins/lifterlms/templates/admin/reporting/tabs/courses/overview.php'), LLMS_Course_Data->get_lesson_completions
```

### System and Environment Information

WordPress 5.9-src
LifterLMS 5.9.0
Windows 10
PHP 8.0.6
MySQL 5.5.5

This issue has be recreated:
+ [x] Locally
+ [ ] On a staging site
+ [ ] On a production website
+ [x] With only LifterLMS and a default theme

### PR DESCRIPTION
**Pull request by reedhewitt**
## Description
If a course does not have any lessons yet, the get_lesson_completions method would cause a MySQL error. With this change, the method will return zero if no lessons exist.

## How has this been tested?
Tested on our production site. The MySQL error no longer occurs and the course report overview shows "0" for lesson completions instead of nothing.

Because this change is so small and the simple, I have not created a test environment and run automated tests.

## Checklist:
- [x] My code has been tested.
- [ ] My code passes all existing automated tests. 
- [ ] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

**Comment by thomasplevy** [comment]
@reedhewitt 

Thanks for the PR -- since we have no associated issue to accompany this PR could you share some reproduction steps to produce the error this PR resolves. 

Could you also address the CS issues identified by phpcs: https://github.com/gocodebox/lifterlms/runs/4401913421?check_suite_focus=true

File: includes/class.llms.course.data.php [hunk 1]
```diff
@@ -164,6 +164,14 @@ public function get_lesson_completions( $period = 'current' ) {
 
 		$lessons = implode( ',', $this->post->get_lessons( 'ids' ) );
 		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+
+		// It is possible to create a course but not have any lessons yet. Return zero now
+		// instead of causing a MySQL error by leaving "post_id IN ( {$lessons} )"
+		// blank in the query.
```
Comment (thomasplevy) [inline includes/class.llms.course.data.php]: We don't have an official CS rule about this but generally we rwrite multiline comments using `/** */` style comments. However, I don't think it's necessary to describe the bug in a comment regardless.
```suggestion
		// Return early for courses without any lessons.
```

Comment (reedhewitt) [reply includes/class.llms.course.data.php]: Ah yes, that makes sense.  I committed your suggested change to the comment.

**Comment by thomasplevy** [comment]
> @reedhewitt
> 
> Thanks for the PR -- since we have no associated issue to accompany this PR could you share some reproduction steps to produce the error this PR resolves.
> 
> Could you also address the CS issues identified by phpcs: https://github.com/gocodebox/lifterlms/runs/4401913421?check_suite_focus=true

@reedhewitt

**Comment by thomasplevy** [comment]
@reedhewitt

File: includes/class.llms.course.data.php [hunk 2]
```diff
@@ -164,6 +164,12 @@ public function get_lesson_completions( $period = 'current' ) {
 
 		$lessons = implode( ',', $this->post->get_lessons( 'ids' ) );
 		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+
+		// Return early for courses without any lessons.
+		if(empty($lessons)){
```
Comment (pondermatic) [inline includes/class.llms.course.data.php:169]: Use [WordPress coding standards for spaces](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#space-usage).
```suggestion
		if ( empty( $lessons ) ) {
```

**Review by pondermatic — CHANGES_REQUESTED** [review]
Please update the [file header DocBlock](https://github.com/gocodebox/lifterlms/blob/trunk/docs/documentation-standards.md#file-headers-1) version tag with a [placeholder](https://github.com/gocodebox/lifterlms/blob/trunk/docs/documentation-standards.md#using-placeholders).
```php
 * @version [version]
```

**Comment by pondermatic** [comment]
Fixes #2012.

**Comment by pondermatic** [comment]
@reedhewitt, please add a [changelog](https://github.com/gocodebox/lifterlms/blob/trunk/docs/documentation-standards.md#changelogs) entry to the [`LLMS_Course_Data::get_lesson_completions()`](https://github.com/reedhewitt/lifterlms/blob/trunk/includes/class.llms.course.data.php#L156) DocBlock.
```
	 * @since [version] Return 0 if there are no lessons.
```

**Comment by thomasplevy** [comment]
@pondermatic at this point just fork it and make the changes and push a new PR (and close this one)

**Comment by pondermatic** [comment]
Closed in favor of PR #2013.

**Pull request closed without merging**